### PR TITLE
[F] Table Content Block

### PR DESCRIPTION
**Pull request by alexgoff**
JIRA: https://jira.lsstcorp.org/browse/EPO-8565

## What this change does ##

Add Table Content Block to Craft and client

## Notes for reviewers ##

Edit or create a new table in Craft, check that the first row of the table is set as the header, check that enabling "isHeader" for any following row's first cell sets it as a header element as well, check that styling and content is as expected.

Include an indication of how detailed a review you want on a 1-10 scale.
- 3 = "I barely need review on this"

## Checklist ##

If any of the following are true, please check them off
- [ ] This is a breaking change: changes in page data (`/src/data/pages/PAGE.json`) or scientific data (anything in `/static/`) and I will notify Product Marketing when it is in prod and user visible.
- [x] This change impacts several investigations (e.g. the change affects reuseed styles, widgets, pages, QAs, utility methods, etc.)
- [ ] This change is isolated to a specific page or Component (i.e. it's a one-off)
- [ ] I don't know what this change does

### CODE REVIEW

File: components/content-blocks/InteractionGroupContainer/styles.ts [hunk 1]
```diff
@@ -5,10 +5,15 @@ export const Heading = styled.h2`
   margin-block-end: ${fluidScale("2em", "1.5em")};
 `;
 export const InteractionGroup = styled.div`
+  --content-block-margin: ${fluidScale(
```
Comment (alexgoff) [inline components/content-blocks/InteractionGroupContainer/styles.ts:8]: trying to standardize this into something reusable

File: components/content-blocks/Table/Table.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      return tableRow.map(({ cellContent = "", rowHeader = false }) => {
+        return {
+          children: cellContent,
+          ...(i > 0 && { isHeader: rowHeader }),
+        };
+      });
+    });
+  };
+
+  const rows = buildRows(Array.from(displayTable) as any);
```
Comment (alexgoff) [inline components/content-blocks/Table/Table.tsx:52]: it's a huge pain getting the types from GQL into something usable

File: components/content-blocks/Table/styles.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  --table-negative: ${fluidScale(
+    "var(--PADDING_LARGE, 100px)",
+    "var(--PADDING_SMALL, 20px)"
+  )};
+
+  margin-inline: calc(var(--table-negative) * -1);
+
+  @media only screen and (min-width: ${token("BREAK_TABLET")}) {
+    --table-negative: 0;
+  }
```
Comment (alexgoff) [inline components/content-blocks/Table/styles.ts:18]: for small screens, tables span to the edge of the screen

File: contexts/Progress.tsx [hunk 4]
```diff
@@ -64,7 +64,8 @@ const getTwoColQuestionEntryIds = (block: object) => {
   const { columns } = block;
 
   const colBlocks = [...columns].map((col) => {
-    return [...col.children];
+    const { children = [] } = col;
```
Comment (alexgoff) [inline contexts/Progress.tsx:67]: encountered a bug here where `children` was undefined

**Review by blnkt — APPROVED** [review]

**Pull request merged**